### PR TITLE
Adding l2_normalize operation

### PR DESCRIPTION
**Pull request by taless474**
This PR adds the functionality of Keras [l2_normalize](https://keras.io/backend/#l2_normalize) except for the case that the input is 3d and no axis is given (l2_normalize3d_flatten). It will be updated when `l2Norm` is added to the blaze_tensor.
It also modifies the transpose operation for the case of `axes=None`. The current version produces the correct response even if the nil value of `axes` is wrapped in a variable.
```py
import numpy as np
from phylanx import Phylanx, PhylanxSession, execution_tree

PhylanxSession.init(1)
	
def variable(value, dtype=None, name=None, constraint=None):
    return execution_tree.var(np.array(value, dtype))
	
@Phylanx
def permute_dimensions_eager(x, pattern):
	return np.transpose(x, pattern)

def permute_dimensions(x, pattern):
	return permute_dimensions_eager.lazy(x, pattern)

def eval(func):
    return func.eval()
	

def test_transpose_operation(axis):
	x_val = [[1,2,3],[4,5,6]]
	t1 = permute_dimensions(variable(x_val),axis)
	z1 = eval(t1)
	return z1
	
test_transpose_operation(None)
```
Working toward #857

### CODE REVIEW

**Comment by hkaiser** [comment]
@taless474 please rebase this branch for it to be merged.

**Comment by hkaiser** [comment]
@taless474 should we merge this even without 3D support?
